charms/kubernetes_snaps.py

```diff
@@ -650,15 +658,34 @@ def install_snap(name: str, channel: str, classic=False, ignore_running=False):
     check_call(cmd)
 
 
-def is_snap_installed(name):
+def is_channel_available(snap_name: str, target_channel: str) -> bool:
+    """
+    Check if the target channel exists for a given snap.
+
+    Args:
+    snap_name (str): Then name of the snap package.
+    target_channel (str): The target channel to find.
+
+    Returns:
+    bool: True if snap channel contains a revision, False otherwise.
+    """
+    cmd = ["snap", "info", snap_name]
+    result = check_output(cmd)
+    output = yaml.safe_load(result)
+    channels = output.get("channels") or {}
+    target = channels.get(target_channel) or None
```

Comment:
I believe you can use the default value in the `get` call. By default, it returns None, so the `target` variable should be fine.
```suggestion
    channels = output.get("channels") or {}
    target = channels.get(target_channel) or None
    return target and target != "--"
```
```suggestion
    channels = output.get("channels", {})
    target = channels.get(target_channel)
```